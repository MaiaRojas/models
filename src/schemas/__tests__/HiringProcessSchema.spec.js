const mongoose = require('mongoose/browser');
const {
  JobOpportunitySchema,
  UserSchema,
  OrganizationSchema,
  HiringProcessSchema,
} = require('../')(mongoose);

describe('HiringProcessSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, HiringProcessSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate ...', (done) => {
    const opportunity = new mongoose.Document({}, JobOpportunitySchema);
    const user = new mongoose.Document({}, UserSchema);
    const organization = new mongoose.Document({}, OrganizationSchema);

    const hiring = new mongoose.Document({
      opportunity: opportunity._id,
      user: user._id,
      organization: organization._id,
      createDate: new Date(),
      acceptedDate: new Date(),
      createProposalDate: new Date(),
      startDate: new Date(),
      salary: 1000,
      active: true,
    }, HiringProcessSchema);

    hiring.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
