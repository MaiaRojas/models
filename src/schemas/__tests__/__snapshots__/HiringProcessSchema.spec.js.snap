// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiringProcessSchema should fail validation when fields missing 1`] = `
Object {
  "active": [ValidatorError: Path \`active\` is required.],
  "createDate": [ValidatorError: Path \`createDate\` is required.],
  "opportunity": [ValidatorError: Path \`opportunity\` is required.],
  "organization": [ValidatorError: Path \`organization\` is required.],
  "user": [ValidatorError: Path \`user\` is required.],
}
`;
